tasks/discover-granules/tests/index.js

```diff
@@ -422,3 +428,45 @@ test.serial('checkGranuleHasNoDuplicate throws an error on an unexpected API lam
       granulesRevert();
     }
   });
+
+test('discover granules sets the GRANULES environment variable and logs the granules', async (t) => {
+    const { event } = t.context;
+
+    event.config.collection.provider_path = '/granules/fake_granules';
+    event.config.provider = {
+      id: 'MODAPS',
+      protocol: 'http',
+      host: '127.0.0.1',
+      port: 3030
+    };
+
+    t.falsy(process.env.GRANULES);
+
+    const loggerInfoSpy = sinon.spy(Logger.prototype, 'info');
+
+    try {
+      await validateConfig(t, event.config);
+
+      await discoverGranules(event);
+
+      t.truthy(process.env.GRANULES);
+
+      const granules = JSON.parse(process.env.GRANULES);
+      t.deepEqual(granules, [
+        'granule-1',
+        'granule-2',
+        'granule-3'
+      ]);
+
+      // Check that the second log.info() call had the granules set
+      const loggerCall = loggerInfoSpy.getCall(1);
```

Comment:
I put this in as a test that we actually set the granules before we did the logging. I'm back and forth though because it's really brittle. If you add another `log.info()` call before this one then this test would fail.